### DISCUSSION TITLE
CI Provider Name Environment Variable

### DISCUSSION THREAD

**Discussion by edcheung1** [cypress-io/cypress#26385] [CI setup] [answered]
Hello,

We have Cypress tests running in our internal CI system. I recently added the environment variable `CYPRESS_CI_BUILD_URL` which successfully adds a link to our internal CI build, but I would like to also customize the displayed CI provider name to something other than `Not available`. Is there another environment variable I can set to change this value?

![image](https://user-images.githubusercontent.com/15271072/229179124-7e5c0799-6ba2-4ccc-bc91-ba65b1bbc149.png)

Thanks!

**Comment by jackkoppa** [comment]
Hi all - on @edcheung1's team, and we've really benefited from being able to link to that build, thank you. Given people may be coming to their failed test run page for the first time, however, we've found it's likely they don't realize they can actually click to the build when what they see is `Not available`

Would love to see if this display value is something that could be configurable; based on where `CYPRESS_CI_BUILD_URL` is setup - https://github.com/cypress-io/cypress/blob/c13079fb1fb1d6d526f6073ca251435a84d4b23e/packages/server/lib/util/ci_provider.js#L141 - I'm guessing it may be a case of adding an additional env var there?

**Comment by idan-marciano-jfrog** [comment]
@edcheung1 @jackkoppa What is the cypress version you use to define this var? i am passing this var to cypress container version 12.0.0 and its not working. the Not available string is not a link. 
As a work around i passed the URL string to another var CI_BUILD_ID thats works but again not as a link.

**Comment by eswanny22** [comment, marked as answer]
Hi everyone, unfortunately, there is no way to change that "Unknown" text. It used to say "Not available" as you've mentioned. If your CI provider isn't on [this list](https://github.com/cypress-io/cypress/blob/a5b0cad9f0b7a49674c79747cc2616e788a3b2da/packages/server/lib/util/ci_provider.js#L91), the text will show as Unknown and there is no env var to change that as of right now.
